operator/internal/manifests/internal/config/loki-config.yaml

```
@@ -201,18 +201,19 @@ limits_config:
 {{- end }}
 {{- end }}{{- end }}
   max_cache_freshness_per_query: 10m
-{{- if .Stack.Limits.Global.IngestionLimits.PerStreamRateLimit }}
-  per_stream_rate_limit: {{ .Stack.Limits.Global.IngestionLimits.PerStreamRateLimit }}MB
+{{- with .Stack.Limits.Global.IngestionLimits.PerStreamRateLimit }}
+  per_stream_rate_limit: {{ . }}MB
 {{- end }}
-{{- if .Stack.Limits.Global.IngestionLimits.PerStreamRateLimitBurst }}
-  per_stream_rate_limit_burst: {{ .Stack.Limits.Global.IngestionLimits.PerStreamRateLimitBurst }}MB
+{{- with .Stack.Limits.Global.IngestionLimits.PerStreamRateLimitBurst }}
+  per_stream_rate_limit_burst: {{ . }}MB
 {{- end }}
   split_queries_by_interval: 30m
-{{- if .Stack.Limits.Global.IngestionLimits.DesiredRate }}
+{{- with .Stack.Limits.Global.IngestionLimits.DesiredRate }}
   shard_streams:
     enabled: true
-    desired_rate: {{ .Stack.Limits.Global.IngestionLimits.DesiredRate }}
+    desired_rate: {{ toBytes . }}
```

(this invalidates the other comment about `toBytes`)

Looking at the Loki sources, it looks like `desired_rate` is using the same type as `per_stream_rate_limit` (`flagext.ByteSize`) and should be able to take a `MB` suffix, even though this is not mentioned in the docs. Both are listed as `<int>` in the docs though as well.

```suggestion
    desired_rate: {{ . }}MB
```

---

Nice find, will use that and get rid of `toBytes`